### DISCUSSION TITLE
Can not edit a text box that was previously part of a group

### DISCUSSION THREAD

**Discussion by sohailahmad07** [fabricjs/fabric.js#9979] [Q&A]
Hi 

so if you group shapes with a textbox and if you ungroup them back you can not edit the textbox 

here is the code pen 

https://codepen.io/Sohail-Ahmad-the-solid/pen/abgoWLo

![Screen Capture_select-area_20240705221232](https://github.com/fabricjs/fabric.js/assets/58299055/6f906d7e-75c3-450e-86b5-f598ec561cdd)

**Comment by asturur** [comment]
I think you linked the wrong codepen.

**Reply by asturur** [reply]
Anyway in case you are wondering, in fabric 6 you do not need to ungroup a text from a group to edit it.
But please link the correct codepen

**Reply by sohailahmad07** [reply]
Hi sorry for the delay 

ok I fix the codepen 

https://codepen.io/Sohail-Ahmad-the-solid/pen/abgoWLo

and yes I know about the new group but i don't want then to move the textbox around its more like the sticky note for miro

**Reply by asturur** [reply]
That codepen is silly, but anyway if you really need to make a group and then add the elements and then remove the group, do a group.remove(ele) before adding it to the canvas, or do a group.destroy() after removing the group from the canvas.

**Reply by sohailahmad07** [reply]
ok here a realistic code pen example and actually what i am doing in my app

user have ability to group and ungroup objects but when i group a textbox and shape and then ungroup after that user can not edit the textbox and group.destory throws and error group.destory is not a function

here codepen

https://codepen.io/Sohail-Ahmad-the-solid/pen/abgoWLo

here video
![Screen Capture_select-area_20240709015002](https://github.com/fabricjs/fabric.js/assets/58299055/0c49a33b-f15b-4eed-9f3f-f3ce5d9b1fe5)

**Reply by sohailahmad07** [reply]
Hi @asturur there is also another issue with group you can check in the same code pen here the issue 

after you ungroup the object selecting all the objects again change its position take a look 

![Screen Capture_select-area_20240709175244](https://github.com/fabricjs/fabric.js/assets/58299055/2c6bd083-e3a7-4331-ad21-29508b2bb88e)
